e2e_tests/tests/cluster/test_master_restart.py

```diff
@@ -75,6 +75,40 @@ def test_master_restart_reattach_recover_experiment_k8s(
     _test_master_restart_reattach_recover_experiment(k8s_managed_cluster, downtime)
 
 
+@pytest.mark.managed_devcluster
+def test_master_agent_restart_reattach_recover_experiment(
+    restartable_managed_cluster: managed_cluster.ManagedCluster,
+) -> None:
+    sess = api_utils.user_session()
+
+    try:
+        # Start an experiment
+        exp_id = exp.create_experiment(
+            sess,
+            conf.fixtures_path("no_op/single-medium-train-step.yaml"),
+            conf.fixtures_path("no_op"),
+            None,
+        )
+
+        # Kill the agent & master
+        restartable_managed_cluster.kill_agent()
```

Comment:
does the order matter here that which one dies first?

Reply:
Yes, if you try to kill the agent after the master, it gives you an error (can't remember what it is off the top of my head though)